lib/openapi_parser/schemas/components.rb

```diff
@@ -24,5 +24,13 @@ class Components < Base
     # @!attribute [r] headers
     #   @return [Hash{String => Header}, nil] header objects
     openapi_attr_hash_object :headers, Header, reference: true
+
+    # @!attribute [r] securitySchemes
+    #   @return [Hash{String => SecurityScheme}, nil]
+    openapi_attr_hash_object :security_schemes, SecurityScheme, reference: false, schema_key: :securitySchemes
+
+    def validate_security_schemes(params, options)
```

Comment:
We need to implement a Security Requirement Object for validation.
The top-level object (OpenAPIParser::Schemas::OpenAPI) have it but 
 we can be overridden it using [security property in Operation Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#operation-object), so both must be checked for correct validation.

RequestOperation is prepared for such a process that does not end with a single object.
Therefore, it is necessary to have an OpenAPI object (or Security Requirement Object and Security Scheme Object only) and a method that compares it with the description of the Operation Object and performs validation. The following is an example of a method that can be used to validate the OpenAPI object.

Translated with www.DeepL.com/Translator (free version)

Reply:
Yes, the `security: []` property is defined for any of the endpoints to allow or deny according to the securityScheme defined (http, oauth, etc)

What method example do you mean?

Reply:
@ota42y How do i get the openapi object in request_operation? I need to write this up. later on i will add any missing ones.

Reply:
@ota42y I've already added `security`

Reply:
@ota42y I need to add this feature in first, OAuth and the rest will be later.

Reply:
@ota42y I found out that the parser in `concerns` does not parse `components`

Reply:
The RequestOperation object doesn't have OpenAPI object now.
The user create it by `request_operation` method so we can pass OpenAPI object to RequestOperation object in this method.
https://github.com/ota42y/openapi_parser/blob/master/lib/openapi_parser/schemas/openapi.rb#L33

Reply:
`I found out that the parser in concerns does not parse components`
It looks to be working fine.
https://github.com/ota42y/openapi_parser/blob/master/spec/openapi_parser/schemas/components_spec.rb

I think it's a bug depending data, what data did you try with?

Reply:
I'm doing it now. Thanks for guiding me. I'm trying to do it as fast as i can. I am just confused on the part you mentioned `OpenAPI object` because there are many things in the code with `OpenAPI`

I've tried add `self` to that `RequestOperation` Object but i only get `#` in result.

Example code i added:
original code:
https://github.com/ota42y/openapi_parser/blob/master/lib/openapi_parser/schemas/openapi.rb#L33-#L35

modified code.
```ruby
    # @return [OpenAPIParser::RequestOperation, nil]
    def request_operation(http_method, request_path)
      OpenAPIParser::RequestOperation.create(http_method, request_path, @path_item_finder, @config, self)
    end
```

Reply:
@ota42y I've made some progress!

![Screenshot 2023-02-22 at 1 28 26 AM](https://user-images.githubusercontent.com/3349747/220417263-c6df2c97-ffc0-4c2e-a046-8631e4d15728.png)